src/thread_task.rs

```diff
@@ -878,14 +943,13 @@ pub unsafe fn rav1d_worker_task(c: &Rav1dContext, task_thread: Arc<Rav1dTaskCont
                             }
                             if (t.sby + 1) < f.sbh {
                                 // add sby+1 to list to replace this one
-                                let next_t_idx = t_idx + 1;
-                                // Rav1dTask is currently 40 bytes, so this
-                                // clone to avoid borrow check issues is cheap
-                                // enough.
-                                let t = t.clone();
-                                let next_t = &mut tasks[next_t_idx];
-                                *next_t = t;
-                                next_t.sby += 1;
+                                let next_t = Rav1dTask {
+                                    next: Default::default(),
+                                    sby: t.sby + 1,
+                                    recon_progress: t.sby + 2,
+                                    deps_skip: AtomicI32::new(t.deps_skip.load(Ordering::Relaxed)),
+                                    ..*t
```

Comment:
```suggestion
                                    sby: t.sby + 1,
                                    recon_progress: t.sby + 2,
                                    ..t.clone()
```

Reply:
We're just cloning `deps_skip`.  It's IMO clearer if we only set the fields we're actually changing.

Reply:
I'd prefer not to here, there's no reason to do a clone just for syntactic sugar.

Reply:
Okay, but the clone and copy here are basically identical.